app/src/main/java/io/github/hidroh/materialistic/DownloadModule.java

```diff
@@ -0,0 +1,54 @@
+package io.github.hidroh.materialistic;
+
+import android.content.Context;
+import dagger.Module;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import javax.inject.Inject;
+import java.io.*;
+
+@Module(library = true, complete = false, includes = NetworkModule.class)
+class DownloadModule {
+    private Call.Factory mCallFactory;
+
+    @Inject
+    DownloadModule(Call.Factory callFactory) {
+        this.mCallFactory = callFactory;
+    }
+
+    void downloadFile(Context context, String url, String mimeType, DownloadModuleCallback callback) {
+        final Request request = new Request.Builder().url(url)
+            .addHeader("Content-Type", mimeType)
+            .build();
+
+        mCallFactory.newCall(request).enqueue(new Callback() {
```

Comment:
It'd be better to call `execute()` instead of `enqueue()` to avoid double callback, since the callback here only processes response on the same thread before relaying to the other callback.

Reply:
But the main thread calls `mDownloadModule.downloadFile`, if I replace `enqueue()` with `execute()` here, I'll get `NetworkOnMainThreadException`, right?

Reply:
Yes you're right I didn't notice this method is to be called on main thread. I'm good with double callback for now, we can replace it with RxJava chain later.